src/bootstrap/compile.rs

```diff
@@ -780,9 +780,21 @@ pub fn rustc_cargo_env(builder: &Builder<'_>, cargo: &mut Cargo, target: TargetS
             cargo.env("LLVM_RUSTLLVM", "1");
         }
         let llvm_config = builder.ensure(native::Llvm { target });
-        cargo.env("LLVM_CONFIG", &llvm_config);
+
+        // HACK: on windows, paths are case insensitive, so case differences cause spurious rebuilds
+        if target.contains("windows") {
+            cargo.env("LLVM_CONFIG", llvm_config.as_os_str().to_ascii_lowercase());
+        } else {
+            cargo.env("LLVM_CONFIG", &llvm_config);
+        }
```

Comment:
It feels like we'll be on a pretty long road to fixing this if we have to make this kind of conversion on every point where Cargo is going to be sensitive to this. Is there a chance this is something we can fix with rust-analyzer or otherwise? It feels a little weird that there's different cases floating around, even if the underlying filesystem is insensitive.